nixos/modules/services/blockchain/bitcoin-core-generic.nix

```diff
@@ -0,0 +1,123 @@
+{ config, options, pkgs, lib
+, name
+, description
+, defaultUser
+, userDescription
+, defaultGroup
+, groupDescription
+, defaultPackage
+, daemonExec
+, cliExec
+, ... }:
+
+with lib;
+
+let
+
+  stateDir = "/var/lib/${name}";
+  pidFile = "${stateDir}/${name}.pid";
+
+  cu = pkgs.coreutils;
+
+  cfg = config.services.${name};
+  opts = options.services.${name};
+
+  configFile = optionalString (!isNull cfg.extraConfig) (" -conf=${pkgs.writeText "${name}.conf" cfg.extraConfig}");
+
+in
+
+{
+
+  options = {
+
+    services.${name} = {
+
+      enable = mkOption {
+        default = false;
+        description = "Enable ${description}";
+        type = types.bool;
+      };
+
+      package = mkOption {
+        default = defaultPackage;
+        description = "Package to use";
+        type = types.package;
+      };
+
+      user = mkOption {
+         default = defaultUser;
+         description = userDescription;
+         type = types.str;
+      };

```

Comment:
since static user/group names are allocated, it is not really configurable. I would just drop these two options here and in every other service.